docs/src/main/mdoc/typeclasses/bimonad.md

```diff
@@ -23,41 +23,41 @@ def f[T[_] : Bimonad, S](fa: T[S]): S
 keep in mind `Bimonad` has its own added laws so something that is both monadic
 and comonadic may not necessarily be a lawful `Bimonad`.
 
-###Eval as a Bimonad
-Eval is a lawful `Bimonad` so you can chain computations (like a `Monad`) and `extract` the result at the end (like a `Comonad`).
+### NonEmptyList as a Bimonad
+NonEmptyList is a lawful `Bimonad` so you can chain computations (like a `Monad`) and `extract` the result at the end (like a `Comonad`).
 
 Here is a possible implementation based on existing monad and comonad:
 ```scala mdoc
 import cats._
 import cats.data._
 import cats.implicits._
 
-implicit def evalBimonad(implicit monad: Monad[Eval], comonad: Comonad[Eval]) =
-  new Bimonad[Eval] {
+implicit def nelBimonad(implicit monad: Monad[NonEmptyList], comonad: Comonad[NonEmptyList]) =
+  new Bimonad[NonEmptyList] {
 
-    //use Eval specific methods for creation and extraction
-    override def pure[A](a: A): Eval[A] =
-      Eval.now(a)
+    //use NonEmptyList specific methods for creation and extraction
+    override def pure[A](a: A): NonEmptyList[A] =
+      NonEmptyList.one(a)
 
-    override def extract[A](x: Eval[A]): A =
-      x.value
+    override def extract[A](fa: NonEmptyList[A]): A =
+      fa.head
 
-    //use the coflatMap from the Eval comonad
-    override def coflatMap[A, B](fa: Eval[A])(f: Eval[A] => B): Eval[B] =
+    //use the coflatMap from the NonEmptyList comonad
+    override def coflatMap[A, B](fa: NonEmptyList[A])(f: NonEmptyList[A] => B): NonEmptyList[B] =
       comonad.coflatMap(fa)(f)
```

Comment:
Actually we can directly call `flatMap` and `coflatMap` directly on the `NonEmptyList` :)

Reply:
I was hopping this shows `flatMap` and `coflatMap` as originating from `monad` and `comonad`. 
Don't you think it's a good idea to delegate to the `monad` and `comonad` variables ?

Reply:
Yes, I do see that now. But then isn't it confusing, because `pure` and `extract` are also defined in `monad` and `comonad`, so we should delegate those as well?

Reply:
> I thought using one and head in the implementation will make this more obvious

I agree, I think this is the right way to do it. So, we can just delegate `flatMap` and `coflatMap` to the list without `monad`/`comonad` and maybe leave `tailRecM` as an exercise for the reader 😉

Anyway just my opinion, not really sure what the best thing to do here is. But I think separating the implementations of `pure` and `extract` makes it seem like they are inconsistent or different from the definitions in `monad` or `comonad`.

Reply:
So the current `tailRecM` implementation for `NonEmptyList[_]` doesn't look trivial .. I mean I fear Its understanding detours too much and I agree it's a good idea to leave it as an exercise 👍 

> Anyway just my opinion, not really sure what the best thing to do here is. But I think separating the implementations of pure and extract makes it seem like they are inconsistent or different from the definitions in monad or comonad.

Yes, I will remove `monad` and `comonad`.